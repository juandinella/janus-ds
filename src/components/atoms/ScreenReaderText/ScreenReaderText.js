import isPropValid from '@emotion/is-prop-valid'
import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'
import styles from './ScreenReaderText.styles'

/**
 * Hidden text just reachable for screen readers
 */

const shouldForwardProp = (prop) => isPropValid(prop)
const StyledScreenReaderText = styled.span.withConfig({ shouldForwardProp })`
  ${styles}
`

const ScreenReaderText = ({ id, children }) => {
  return (
    <StyledScreenReaderText id={id} data-testid={id}>
      {children}
    </StyledScreenReaderText>
  )
}

ScreenReaderText.displayName = 'ScreenReaderText'

ScreenReaderText.propTypes = {
  /** Component's id */
  id: PropTypes.string,
  /** Component's content */
  children: PropTypes.string.isRequired,
}

export default ScreenReaderText
