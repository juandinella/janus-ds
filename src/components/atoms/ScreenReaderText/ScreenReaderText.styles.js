import { css } from 'styled-components'

export default css`
  border: 0;
  padding: 0;
  margin: 0;
  position: absolute;
  height: 1px;
  width: 1px;
  overflow: hidden;
  clip: rect(1px 1px 1px 1px);
  clip: rect(1px, 1px, 1px, 1px);
  clip-path: inset(50%);
  white-space: nowrap;
`
