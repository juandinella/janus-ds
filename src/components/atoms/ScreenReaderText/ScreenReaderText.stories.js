import ScreenReaderText from './ScreenReaderText'

import { getTemplate } from '../../../helpers/storybook'

const Template = getTemplate(ScreenReaderText)

export default {
  title: 'Atoms/ScreenReaderText',
  component: ScreenReaderText,
  args: {
    children: 'Hidden text',
  },
  argTypes: {},
}

export const Default = Template.bind({})
