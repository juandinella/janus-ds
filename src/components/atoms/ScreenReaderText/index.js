import ScreenReaderText from './ScreenReaderText'

export default ScreenReaderText
