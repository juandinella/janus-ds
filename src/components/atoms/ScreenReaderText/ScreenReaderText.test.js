import initStoryshots from '@storybook/addon-storyshots'
import { render, cleanup, screen } from '@testing-library/react'
import React from 'react'
import ScreenReaderText from './ScreenReaderText'

initStoryshots({
  storyKindRegex: /^Atoms\/ScreenReaderText$/,
})

const componentProps = {
  id: 'unique_id',
  children: 'Go!',
}

afterEach(cleanup)

describe('<ScreenReaderText />', () => {
  test('should render component correctly', () => {
    render(<ScreenReaderText {...componentProps} />)
    const comp = screen.getByTestId(componentProps.id)
    expect(comp).toBeInTheDocument()
  })

  test('renders the content correctly', () => {
    render(<ScreenReaderText {...componentProps} />)
    const link = screen.getByTestId(componentProps.id)
    expect(link).toHaveTextContent(componentProps.children)
  })
})
