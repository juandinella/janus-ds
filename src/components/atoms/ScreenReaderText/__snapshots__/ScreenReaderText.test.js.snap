// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Atoms/ScreenReaderText Default 1`] = `
<div
  style={
    {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
      "position": "relative",
    }
  }
>
  <span
    className="sc-90c4yq-0 bmmsAZ"
  >
    Hidden text
  </span>
</div>
`;
